server/svix-server/src/main.rs

```diff
@@ -115,5 +123,80 @@ async fn main() {
         None => {}
     };
 
+    if let Some(wait_for_seconds) = args.wait_for {
+        let pg = &cfg.db_dsn;
+
+        let pg = url::Url::parse(pg).expect("Invalid PostgreSQL DNS");
+
+        let pg_host = pg.host().expect("Expected PostgreSQL host");
+        let pg_port = pg.port().unwrap_or(5432);
+
+        let pg_host: IpAddr = match pg_host {
+            url::Host::Domain(s) => {
+                let resolver = TokioAsyncResolver::new(
+                    ResolverConfig::default(), ResolverOpts::default(), TokioHandle)
+                    .expect("Could not set up DNS resolution");
+
+                let resp = resolver.lookup_ip(s).await.expect("DNS error");
+                resp.iter()
+                    .next()
+                    .expect("Expected at least one IP address")
+            }
+            url::Host::Ipv4(ipv4_addr) => ipv4_addr.into(),
+            url::Host::Ipv6(ipv6_addr) => ipv6_addr.into(),
+        };
+
+        let sleep = tokio::time::sleep(Duration::from_secs(wait_for_seconds));
+        tokio::pin!(sleep);
+        loop {
+            tokio::select! {
+                _ = tokio::net::TcpStream::connect((pg_host, pg_port)) => {
+                    break;
+                },
+                _ = &mut sleep => {
+                    panic!("Waiting for databases timed out");
+                }
+            }
+        }
+
+        let redis = cfg.redis_dsn.as_deref();
+
+        if let Some(redis) = redis {
+            let redis = url::Url::parse(redis).expect("Invalid Redis DSN");
+
+            let redis_host = redis.host().expect("Exepected Redis host");
+            let redis_port = redis.port().unwrap_or(6379);
+
+            let redis_host: IpAddr = match redis_host {
+                url::Host::Domain(s) => {
+                let resolver = TokioAsyncResolver::new(
+                    ResolverConfig::default(), ResolverOpts::default(), TokioHandle)
+                    .expect("Could not set up DNS resolution");
+
+                let resp = resolver.lookup_ip(s).await.expect("DNS error");
+                resp.iter()
+                    .next()
+                    .expect("Expected at least one IP address")
+
+                }
+                url::Host::Ipv4(ipv4_addr) => ipv4_addr.into(),
+                url::Host::Ipv6(ipv6_addr) => ipv6_addr.into(),
+            };
```

Comment:
Resolver code should be a helper function, it's exactly the same as the one above.